### PR TITLE
Add test for IPv6 Uris

### PR DESCRIPTION
**Pull request by albe**
This shows a b/c break introduced with #386

### CODE REVIEW

**Comment by albe** [comment]
```
There was 1 failure:
1) GuzzleHttp\Tests\Psr7\UriTest::testIPv6Host
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-[2a00:f48:1008::212:183:10]
+[2a00:f48:1008::212:183
/home/travis/build/guzzle/psr7/tests/UriTest.php:699
FAILURES!
Tests: 837, Assertions: 1947, Failures: 1, Skipped: 1.
```

File: tests/UriTest.php [hunk 1]
```diff
@@ -692,6 +692,15 @@ public function testInternationalizedDomainName()
...
         self::assertSame('яндекaс.рф', $uri->getHost());
     }
+
+    public function testIPv6Host()
+    {
+        $uri = new Uri('https://[2a00:f48:1008::212:183:10]');
+        self::assertSame('[2a00:f48:1008::212:183:10]', $uri->getHost());
+
+        $uri = new Uri('http://[2a00:f48:1008::212:183:10]');
+        self::assertSame('[2a00:f48:1008::212:183:10]', $uri->getHost());
```
Comment (Nyholm) [inline tests/UriTest.php:702]: We dont need these. They are the same.

Comment (albe) [reply tests/UriTest.php:702]: Likely, though I'm rather on the safe side with small but common differences (http vs http*s*), especially when using custom parsing logic.

**Comment by Nyholm** [comment]
FYI #403

**Comment by GrahamCampbell** [comment]
Thanks for providing those test cases. We can continue discussion on #403. :)

**Pull request closed without merging**